### PR TITLE
refactor: make renderVideo depend on project file instead of vite config, let user pass vite port and disable meta file writes

### PR DESCRIPTION
**Pull request by justusmattern27**

### CODE REVIEW

**Comment by justusmattern27** [comment]
Up until now, renderVideo required a path to the vite config file (./vite.config.ts). However, in practice, we barely relied on the vite config:

- we passed arguments like output file names that could be specified in vite.config.ts through renderVideo() even though they could also be specified in vite.config.ts
- we ignored some arguments in vite.config.ts (e.g. we hardcoded ./src/project.ts)

This PR lets renderVideo point to the projectFile instead of vite.config.ts. Now, the UI editor relies on the vite config, and renderVideo purely on arguments the user passes. This makes things less ambiguous.

This also adds some features:
- We can now have multiple video templates in one Revideo project. When you add `projects: ["./src/project.ts",  "./src/project2.ts"]` in vite.config.ts, the UI will let you select which project you want to preview, and you can simply change the project path in renderVideo to select different templates
- not depending on vite.config.ts made it easy to add an output folder option, which we need for Lambda deployment

Other stuff:
- I worked around using the normalize flag in ffmpeg to support more ffmpeg versions
- you can pass the base vite port you want to use via renderVideo
- you have the option to not write to metafiles via an environment variable

File: packages/ffmpeg/src/ffmpeg-exporter-server.ts [hunk 1]
```diff
@@ -251,7 +251,7 @@ export class FFmpegExporterServer {
 
       command
         .complexFilter([
-          `amix=inputs=${this.audioFilenames.length}:duration=longest:normalize=0`,
+          `amix=inputs=${this.audioFilenames.length}:duration=longest,volume=${this.audioFilenames.length}`,
```
Comment (hkonsti) [inline packages/ffmpeg/src/ffmpeg-exporter-server.ts:254]: Not sure if this does the same thing. Let's test it a little more.

File: packages/renderer/server/render-video.ts [hunk 2]
```diff
@@ -52,6 +53,8 @@ export interface RenderVideoSettings {
   workers?: number;
   dimensions?: [number, number];
   logProgress?: boolean;
+  outDir?: string;
```
Comment (hkonsti) [inline packages/renderer/server/render-video.ts:56]: This object is getting quite messy. Let's restructure it and do a breaking release. I'm also not sure whether I like outDir and name being two different params. Maybe we can put them into an object like `{ out: { file, workingDirectory} }` or something like that.

File: packages/renderer/server/render-video.ts [hunk 3]
```diff
@@ -60,18 +63,25 @@ export interface RenderVideoSettings {
...
+  const resolvedProjectPath = path.join(process.cwd(), projectFile);
   const [browser, server] = await Promise.all([
     puppeteer.launch({headless: true, ...settings.puppeteer}),
     createServer({
-      configFile: resolvedConfigPath,
+      configFile: false,
       server: {
         port: fixedPort,
+        hmr: false,
+        host: settings.viteServerHost,
```
Comment (hkonsti) [inline packages/renderer/server/render-video.ts:79]: We could expose the entire server objects in the settings by adding vite to `RenderVIdeoSettings` the same way we added `puppeteer?: BrowserLaunchArgumentOptions`.

File: packages/template/src/render.ts [hunk 4]
```diff
@@ -4,10 +4,10 @@ async function render() {
   console.log('Rendering video...');
 
   const file = await renderVideo(
-    './vite.config.ts',
+    './src/project.ts',
     {fill: 'orange'},
     () => {},
-    {logProgress: true},
+    {logProgress: true, outDir: 'out', name: 'result'},
```
Comment (hkonsti) [inline packages/template/src/render.ts:10]: This should have a default and not be defined in here IMO.

File: packages/vite-plugin/src/partials/meta.ts [hunk 5]
```diff
@@ -47,11 +47,13 @@ export default meta;
         }
 
         timeStamps[source] = Date.now();
-        await fs.promises.writeFile(
-          source,
-          JSON.stringify(data, undefined, 2),
-          'utf8',
-        );
+        if (!process.env.META_NOWRITE) {
```
Comment (hkonsti) [inline packages/vite-plugin/src/partials/meta.ts:50]: I think we should make the name more descriptive maybe `DONT_WRITE_TO_META_FILES` or something like that

**Pull request merged**